sql/information_schema/information_schema.go

```diff
@@ -1142,6 +1182,57 @@ func referentialConstraintsRowIter(ctx *Context, c Catalog) (RowIter, error) {
 	return RowsToRowIter(rows...), nil
 }
 
+// schemaPrivilegesRowIter implements the sql.RowIter for the information_schema.SCHEMA_PRIVILEGES table.
+func schemaPrivilegesRowIter(ctx *Context, c Catalog) (RowIter, error) {
+	var rows []Row
+	privSet, _ := ctx.GetPrivilegeSet()
+	if privSet.Has(PrivilegeType_Select) || privSet.Database("mysql").Has(PrivilegeType_Select) {
+		var users = make(map[*mysql_db.User]struct{})
+		db, err := c.Database(ctx, "mysql")
+		if err != nil {
+			return RowsToRowIter(rows...), nil
+		}
```

Comment:
The error handling in this function wasn't what I was expecting... is there a reason we need to diverge from returning any errors and return partial results? If so, that would be a good comment to add to help future code readers understand why this code is different than the normal error handling pattern.